### PR TITLE
Support Bucket policy

### PR DESCRIPTION
**Pull request by RedbackThomson**
Adds support for creating, updating and delete bucket Policy. This field is simply a JSON string that applies IAM permissions to objects in the bucket. 

This is the first `Put*` field implemented so far that has a separate `Delete*` call, so the logic for `syncBucket` now checks accordingly.

### CODE REVIEW

**Comment by RedbackThomson** [comment]
While writing this I also realised that the e2e tests do not clean up their buckets if the assertions fail. That will need some work to rewrite.

File: pkg/resource/bucket/hook.go [hunk 1]
```diff
@@ -62,6 +63,11 @@ func (rm *resourceManager) createPutFields(
 			return err
 		}
 	}
+	if r.ko.Spec.Policy != nil {
+		if err := rm.syncPolicy(ctx, r); err != nil {
+			return err
+		}
+	}
```
Comment (A-Hilaly) [inline pkg/resource/bucket/hook.go:70]: What if the `spec.policy` was deleted from the CR? `rm.deletePolicy(ctx, r)` will never be called i assume?

Comment (RedbackThomson) [reply pkg/resource/bucket/hook.go:70]: This is in the `createPutFields` method, which is called after `CreateBucket`. For updates, we invoke `syncPolicy` through `customUpdateBucket`, which checks for `delta.DifferentAt("Spec.Policy")`.

**Pull request merged**